source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -64,13 +120,39 @@ bool LocalRateLimiterImpl::requestAllowed() const {
     synchronizer_.syncPoint("allowed_pre_cas");
 
     // Loop while the weak CAS fails trying to subtract 1 from expected.
-  } while (!tokens_.compare_exchange_weak(expected_tokens, expected_tokens - 1,
-                                          std::memory_order_relaxed));
+  } while (!tokens.tokens.compare_exchange_weak(expected_tokens, expected_tokens - 1,
+                                                std::memory_order_relaxed));
 
   // We successfully decremented the counter by 1.
   return true;
 }
 
+bool LocalRateLimiterImpl::requestAllowed(
+    std::vector<Envoy::RateLimit::LocalDescriptor> route_descriptors) const {
+  const Envoy::RateLimit::LocalDescriptor* descriptor = findDescriptor(route_descriptors);
+  if (descriptor == nullptr) {
+    return requestAllowedHelper(tokens_);
+  }
+  auto it = tokens_per_descriptor_.find(*descriptor);
```

Comment:
nit: inline this in the next line. Also, can we somehow avoid a 2nd has map lookup here? When we find descriptor above can we somehow have a reference to what we need? Why do we need the map at all? I think now it can be removed entirely along with all of the hash/equality code?

Reply:
I wonder how we should treat extraneous entries in the descriptor. Do we want {a:b, c:d} descriptor to match {a:b} entry in the config, or should it be exact match?

Reply:
That's a good question. I'm not sure. Initially I would do exact match and clearly document it and we can always make it configurable later?

Reply:
It might help with choosing a data structure. I guess keeping it as a map is fine for now. I'll add comments.

Reply:
I think we can add a reference to the state from local descriptor (to make it copyable).